### DISCUSSION TITLE
VSCode Remote SSH setup : How To Twinny ?

### DISCUSSION THREAD

**Discussion by lateo-net** [rjmacarthy/twinny#345] [Q&A]
Currently playing with Ollama at work (air gaped network).
Twinny looks nice and would be a killer feature for the team, but I fail to set it up.

What we have : 
- windows clients with VSCodium + RemoteSSH extension
- a sweet linux development server with loads of CPU cores and running Ollama on GPU
- windows users are connected to the server through VSCodium so they can use all the linux goodness while keeping the weak windows machines clean and unbloated.

What i fail to understand : should the extension be installed server side (with localhost configuration ?) or client side (remote connexion from the windows machine) ? We can have both when using RemoteSSH and i couldn't find a "twiny design sheet", so it isn't clear to me.

What would be nice : a way to debug the connexion to ollama api.

TY,

**Comment by rjmacarthy** [comment]
Hey thanks for the question.  I guess client installs the extension and points to the server.  This is a personal/community project and technical support is it not offered on an individual basis.  If you manage to get something working please share for others.  Many thanks!

**Comment by lateo-net** [comment]
Ok I took some time to look at it before getting out of work. Some quick notes.

- From SSH (distant plugin install), Twinny 3.19.x for Linux: error, "cannot fetch".
- From Windows (local plugin install), Twinny 3.20.4 for Windows : 
    - Chat works fine. 
    - If I go back to the distant (linux) Window : cannot use local install from distant window (plugin greyed, VSC suggests installing plugin to the SSH session)
- Fun stuff (looks like a bug) : 
    - Conversation generated in local window appears in distant window.
    - If I then try to continue the chat in distant Window : error, cannot fetch.

I'll add more info after upgrading the distant plugin to the latest version.

**Reply by lateo-net** [reply]
So, the first answer I got is : 
- if I want my users to be able to use Twinny while working with the dev server and have it help with coding, the extension has to be installed in the user's profile on the server. But this setup doesn't work so far.
- if I want my users to play with ollama only to ask some random question (chat), they can do it if they use Twinny from a classic local VSC session.

Or they could use 2 VSC windows (1 local to use Twinny & 1 distant to do some work), but that's not really an option.
